pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -271,6 +273,8 @@ private void init() {
         _allColumns.add(columnName);
       }
     }
+    //set the table name
+    _tableName = _segmentMetadataPropertiesConfiguration.getString(V1Constants.MetadataKeys.Segment.TABLE_NAME);
```

Comment:
The init() method seems to be private and called only from one of the constructors. Maybe we can make _tableName as final and initalize it in the constructor? Even move the logic from init to the constructor and make more vars final, but that will be a cleanup operation, so your choice.